tests/testsuite/registry.rs

```diff
@@ -30,15 +29,15 @@ fn test_server_stops() {
 #[cargo_test]
 fn simple_http() {
     let _server = setup_http();
-    simple();
+    simple(true);
 }
 
 #[cargo_test]
 fn simple_git() {
-    simple();
+    simple(false);
 }
 
-fn simple() {
+fn simple(is_http: bool) {
```

Comment:
I feel it a bit confusing. By just looking at code, I don't know why this `is_http` is needed. Apparently there is not different between each branch. Is there a way to document it, or this should be fixed in snapbox or go the other way round?

Reply:
Currently, if a function contains `str![]` and is called twice, the content will become something like `str![[...] [...]]`. 

This PR introduces an argument `is_http` to handle `str![]` separately, allowing it to be filled with the correct content (even though the current testsuite doesn't show the difference).

If appending multiple times is not the expected behavior, I think this should be fixed in snapbox.

Alternatively, it's possible to eliminate the need for the `is_http` argument by avoiding `str![]` altogether.

Reply:
I think `tests/testsuite/list_availables.rs` is also worth mentioning here. It reuses a function for assertions with distinct commands multiple times.

Reply:
Thanks for the reply!

I still found it cumbersome to understand. Maybe we should parameterize on inline snapshot itself. That may look like

```rust
fn simple_git() {
    simple(str![[r#"
[UPDATING] `dummy-registry` index
[LOCKING] 2 packages to latest compatible versions
[DOWNLOADING] crates ...
[DOWNLOADED] bar v0.0.1 (registry `dummy-registry`)
[CHECKING] bar v0.0.1
[CHECKING] foo v0.0.1 ([ROOT]/foo)
[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s

"#]], str[[r#"
[CHECKING] bar v0.0.1
[CHECKING] foo v0.0.1 ([ROOT]/foo)
[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s

"#]]);
}
```


(Note that we have two shared assertions hence two arguments)

What do you think?

Reply:
Yeah, this also make sense to me! Thanks!